File: kora-app-annotation-processor/src/main/java/ru/tinkoff/kora/kora/app/annotation/processor/ResolvedGraph.java
package ru.tinkoff.kora.kora.app.annotation.processor;

import ru.tinkoff.kora.kora.app.annotation.processor.component.ResolvedComponents;
import ru.tinkoff.kora.kora.app.annotation.processor.declaration.ComponentDeclarations;

import javax.lang.model.element.TypeElement;
import java.util.List;

public record ResolvedGraph(TypeElement root, List<TypeElement> allModules, ComponentDeclarations declarations, ResolvedComponents components) {}

File: kora-app-symbol-processor/src/main/kotlin/ru/tinkoff/kora/kora/app/ksp/ResolvedGraph.kt
package ru.tinkoff.kora.kora.app.ksp

import com.google.devtools.ksp.symbol.KSClassDeclaration
import ru.tinkoff.kora.kora.app.ksp.component.ResolvedComponents
import ru.tinkoff.kora.kora.app.ksp.declaration.ComponentDeclarations

class ResolvedGraph(
    val root: KSClassDeclaration,
    val allModules: List<KSClassDeclaration>,
    val declarations: ComponentDeclarations,
    val components: ResolvedComponents
) {}
